Improve Hero Section

daganomri opened the pull request:
## Changes Include:
* Reduced image sizes (png to jpg for 80% reduction)
* Making hero section 100vh with scroll down gif on small devices
* Get rid of horizontal scrollbar
* Add will-change to boxes to remove text blur while transitioning
* Add alt text to images

daganomri committed:
changes include:
* reduced image sizes (png to jpg for 80% reduction)
* making hero section 100vh with scroll down gif on small devices
* get rid of horizontal scrollbar
* add will-change to boxes to remove text blur while transitioning
* add alt text to images

guytepper commented on styles.css:
```diff
@@ -95,6 +111,18 @@ h4 {
+}
+@media (min-width: 1330px) {
+  .scroll-down {
+    display: none;
```
Why does it need to be hidden on large viewports?

daganomri replied:
Since it doesn't look good on them. Also, there's the scrollbar to indicate scrolling. I changed it so on medium screens you can see the rest of the content.

guytepper commented on styles.css:
```diff
@@ -95,6 +111,18 @@ h4 {
+  flex-basis: 5%;
+  flex-shrink: 0.5;
+  animation: fadeIn 9.5s ease-out;
+}
```
Spacing
```suggestion
}

```

guytepper commented on styles.css:
```diff
@@ -95,6 +111,18 @@ h4 {
 }
 
+.scroll-down {
+  max-height: 10%;
```
Why not setting the width to something fixed, like `30px`? I think it looks well for all viewports

daganomri replied:
30px looks small on big screens. I actually changed it so the scroll gif only shows on small screens. On big screens there's a scrollbar and on middle screens you can see the rest of the content. I think 10% with the combination of flex-basis and flex-shrink results in a size that is always alright.

guytepper commented on styles.css:
```diff
@@ -95,6 +111,18 @@ h4 {
+  max-height: 10%;
+  flex-basis: 5%;
+  flex-shrink: 0.5;
+  animation: fadeIn 9.5s ease-out;
```
It feels too bright to me, let's make it grey? :)
```suggestion
  filter: contrast(0%);
  animation: fadeIn 9.5s ease-out;
```

daganomri replied:
Done. looks better.

guytepper commented on styles.css:
```diff
@@ -29,11 +29,27 @@ body {
 
 .header {
   width: 100%;
+  height: 100vh;
```
It works well for mobile, but for some large viewports the margins seems too big.

<img width="530" alt="Screen Shot 2021-02-15 at 7 53 13" src="https://user-images.githubusercontent.com/13344923/107910353-e8508000-6f62-11eb-95d5-3f75b00fc292.png">

I think a possible solution is to shorten the height value for those - and show a bit of the content below, similar to the current implementation (maybe show less content though):

<img width="530" alt="image" src="https://user-images.githubusercontent.com/13344923/107910655-86444a80-6f63-11eb-8535-5a3e4749a312.png">

wdyt?

daganomri replied:
I think you're right. I'm working now on only having the scroll locked on big screens and small ones. In the middle tier you see the rest of the content and can scroll to it.

guytepper commented on styles.css:
```diff
@@ -29,11 +29,27 @@ body {
-  overflow: none;
+  overflow-y: scroll;
+  overflow-x: hidden;
+  scroll-behavior: smooth;
```
What's the effect of this property?

daganomri replied:
overflow-x: hidden; is to get rid of the weird horizontal scroll.
overflow-y: scroll; is so the scrollbar will show on desktop even when there is no scroll available.
scroll-behavior: smooth; I don't think does anything in this context, it stayed from my experiments with locking the scroll.

guytepper reviewed (commented):
Great work, thank you! I've added some comments :)

daganomri committed:
Fixed according to notes.
Also added way to lock screen only on first render.
Previous version fired off the lock on every resize.

guytepper committed:
Update alt tags

Since screen reader might read those, I updated the text to Hebrew and make applied it only to one screenshot.

guytepper committed:
Optimized GIF size
